### PR TITLE
[MRG] add information about versioning, migrations, etc to the docs

### PR DESCRIPTION
**Pull request by ctb**
Upgrade documentation in preparation for v4.

ref #1033, https://github.com/dib-lab/sourmash/issues/655, #1069, #1145, #1128 

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
@luizirber the overall shape of the text is at a resting place, and I welcome your thoughts on what topics I'm missing! (I've yet to review the actual formatting.)

File: doc/index.md [hunk 1]
```diff
@@ -108,6 +108,25 @@ microbial genomes with about 200 MB of disk space and about 10 GB of
 RAM. Typically a metagenome classification takes about 1 minute on a
 laptop.
 
+## Sourmash versioning
```
Comment (luizirber) [inline doc/index.md]:
```suggestion
## sourmash versioning
```

File: doc/index.md [hunk 2]
```diff
@@ -108,6 +108,25 @@ microbial genomes with about 200 MB of disk space and about 10 GB of
...
+* Major releases, like v4.0.0, may break backwards compatibility at
+  the command line as well as top-level Python/Rust APIs.
+* Minor releases, like v4.1.0, will remain backwards compatible but
+  may introduce significant new features.
+* Patch releases, like v4.1.1, are for minor bug fixes; full backwards
+  compatibility is retained.
+
+If you are relying on sourmash in a pipeline or application, we
+suggest specifying your version requirements at the major release,
+e.g. in conda you would specify `sourmash>=3,<4`.
```
Comment (luizirber) [inline doc/index.md:126]: ooh, very good advice!

**Comment by luizirber** [comment]
> @luizirber the overall shape of the text is at a resting place, and I welcome your thoughts on what topics I'm missing! (I've yet to review the actual formatting.)

I'm pretty happy with it as-is, I think it touches all the important points!

**Comment by ctb** [comment]
Ready for review!

**Review by luizirber — APPROVED** [review]

**Pull request merged**